tests/pytests/test_vecsim.py

```diff
@@ -792,3 +792,53 @@ def test_wrong_vector_size(env):
     assertInfoField(env, 'idx', 'num_docs', '2')
     assertInfoField(env, 'idx', 'hash_indexing_failures', '4')
     env.expect('FT.SEARCH', 'idx', '*=>[KNN 6 @v $q]', 'NOCONTENT', 'PARAMS', 2, 'q', np.ones(dimension, 'float32').tobytes()).equal([2L, '1', '4'])
+
+def test_hybrid_query_cosine(env):
+    conn = getConnectionByEnv(env)
+    dim = 4
+    index_size = 6000 * env.shardsCount
+    conn.execute_command('FT.CREATE', 'idx', 'SCHEMA', 'v', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32',
+                         'DIM', dim, 'DISTANCE_METRIC', 'COSINE', 't', 'TEXT')
+
+    p = conn.pipeline(transaction=False)
+    for i in range(1, index_size+1):
+        first_coordinate = np.float32([float(i)/index_size])
+        vector = np.concatenate((first_coordinate, np.ones(dim-1, dtype='float32')))
+        conn.execute_command('HSET', i, 'v', vector.tobytes(), 't', 'text value')
+    p.execute()
+
+    query_data = np.ones(dim, dtype='float32')
+
+    expected_res_ids = set([str(index_size-i) for i in range(10)])
+    res = conn.execute_command('FT.SEARCH', 'idx', '(text value)=>[KNN 10 @v $vec_param]',
+           'SORTBY', '__v_score',
+           'PARAMS', 2, 'vec_param', query_data.tobytes(),
+           'RETURN', 1, '__v_score')
+    prefix = "_" if env.isCluster() else ""
+    env.assertEqual(env.cmd(prefix+"FT.DEBUG", "VECSIM_INFO", "idx", "v")[-1], 'HYBRID_BATCHES')
+    # The order of ids is not accurate due to floating point numeric errors, but the top k should be
+    # the last 10 ids, and their scores should all be in the right COSINE range.
+    actual_res_ids = set([res[1:][2*i] for i in range(10)])
+    actual_res_scores = [res[1:][2*i + 1][1] for i in range(10)]
+    env.assertEqual(actual_res_ids, expected_res_ids)
+    for score in actual_res_scores:
+        env.assertTrue(0.0 <= float(score) <= 2.0)
```

Comment:
you can drop this. it is meaningless. if we want to verify the correctness of the scores (e.g. between [0,2]) we should do it in the library